### PR TITLE
[CONTRIB][MAINTENANCE] Custom query expectation and editing query.template_values metric

### PR DESCRIPTION
**Pull request by mantasmy**
Changes proposed in this pull request:
- add new expectation **expect_queried_custom_query_results_to_be_custom**. Custom query can be passed as user_query inside template_dict using **query.template_values** metric and tested if output count matches the value.
```
"template_dict": {
                 "user_query": "select * from {active_batch}"
                 },
"value": 0
```
- adjust **query.template_values**. Overwriting active_batch inside dict was added.

### CODE REVIEW

**Comment by mantasmy** [comment]
@cla-bot check

File: contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    examples = [
+        {
+            "data": [
+                {
+                    "dataset_name": "test",
+                    "data": {"col1": [1, 2, 3, 4, 5, 5], "col2": ["10", 3, 4, 4, 5, 5]},
+                }
+            ],
+            "tests": [
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py:78]: At the same nesting level as tests, add `"only_for": ["sqlite", "bigquery", "redshift", "snowflake"],`

File: contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                {
+                    "title": "basic_positive_test",
+                    "exact_match_out": False,
+                    "include_in_gallery": True,
+                    "in": {
+                        "template_dict": {"user_query": "select * from test"},
+                        "value": 6,
+                    },
+                    "out": {"success": True},
+                    "only_for": ["sqlite"],
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py]: Delete this `only_for` line

File: contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                {
+                    "title": "basic_negative_test",
+                    "exact_match_out": False,
+                    "include_in_gallery": True,
+                    "in": {
+                        "template_dict": {"user_query": "select * from test"},
+                        "value": 1,
+                    },
+                    "out": {"success": False},
+                    "only_for": ["sqlite"],
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py]: Delete this `only_for` line

File: contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            "success": success,
+            "result": {"observed_value": query_result},
+        }
+
+    examples = [
+        {
+            "data": [
+                {
+                    "dataset_name": "test",
+                    "data": {"col1": [1, 2, 3, 4, 5, 5], "col2": ["10", 3, 4, 4, 5, 5]},
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_queried_custom_query_to_return_num_rows.py]: Consider changing the `"10"` to `10`. Spark fails with `TypeError("field col2: Can not merge type <class 'pyspark.sql.types.StringType'> and <class 'pyspark.sql.types.LongType'>")` when trying to get a test validator.

Ditto for Trino `ProgrammingError('(trino.exceptions.TrinoUserError) TrinoUserError(type=USER_ERROR, name=TYPE_MISMATCH, message="line 2:2: Values rows have mismatched types: row(integer, varchar(2)) vs row(integer, integer)", query_id=20230321_152044_00366_yc6mi)')`

**Review by kenwade4 — CHANGES_REQUESTED** [review]
Thanks for contributing this new Expectation @mantasmy! There are some minor changes requested (to allow more backends to work with this Expectation).

In addition to the changes requested by line number, here's a bonus for postgresql support

```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) subquery in FROM must have an alias
LINE 3:             from (select * from test)
                         ^
HINT:  For example, FROM (SELECT ...) [AS] foo.

[SQL:
            select count(1)
            from (select * from test)
            ]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

It's possible that adding an alias will also not break the backends that this Expectation is currently working with (sqlite, bigquery, redshift, snowflake). What do you think @itaise?

**Comment by mantasmy** [comment]
@kenwade4 Thanks! Did the adjustments as requested. 
In regards of alias. I have added it for now and tested in Snowflake - it works well (tested query in SF UI with alias, not expectation itself). Not sure about rest of the backends.
But @itaise has special handling for alias in **query_temp_values.py** (metric) @ line 68, 82. So maybe it won't work well for all of the backends?

**Review by kenwade4 — APPROVED** [review]
Good stuff. The alias change and removing quotes from that number in sample data worked (postgresql and trino)

**Comment by kenwade4** [comment]
@mantasmy you may want to try to get spark working and submit a separate PR in the future. Thanks again.

**Pull request merged**